pkg/apis/core/validation/validation.go

```diff
@@ -3720,10 +3756,69 @@ func ValidatePodCreate(pod *core.Pod, opts PodValidationOptions) field.ErrorList
 	if len(pod.Spec.EphemeralContainers) > 0 {
 		allErrs = append(allErrs, field.Forbidden(fldPath.Child("ephemeralContainers"), "cannot be set on create"))
 	}
+	allErrs = append(allErrs, ValidateSeccompAnnotationsAndFields(pod.ObjectMeta, &pod.Spec, field.NewPath(""))...)
+
+	return allErrs
+}
+
+// ValidateSeccompAnnotationsAndFields iterates through all containers and ensure that when both seccompProfile and seccomp annotations exist they match.
+func ValidateSeccompAnnotationsAndFields(objectMeta metav1.ObjectMeta, podSpec *core.PodSpec, fldPath *field.Path) field.ErrorList {
```

Comment:
unexport unless used outside this package:
```suggestion
func validateSeccompAnnotationsAndFields(objectMeta metav1.ObjectMeta, podSpec *core.PodSpec, fldPath *field.Path) field.ErrorList {
```